msgraph-developer-proxy/ConsoleLogger.cs

```diff
@@ -7,30 +7,315 @@ namespace Microsoft.Graph.DeveloperProxy;
 
 public class ConsoleLogger : ILogger {
     private readonly ConsoleColor _color;
+    private readonly LabelMode _labelMode;
+    private readonly LogLevel _logLevel;
+    private readonly string _boxTopLeft = "\u250c ";
+    private readonly string _boxLeft = "\u2502 ";
+    private readonly string _boxBottomLeft = "\u2514 ";
+    // used to align single-line messages
+    private readonly string _boxSpacing = "  ";
 
-    public ConsoleLogger() {
+    private static readonly object lockObject = new object();
+
+    public ConsoleLogger(ProxyConfiguration configuration) {
         _color = Console.ForegroundColor;
+        _labelMode = configuration.LabelMode;
+        _logLevel = configuration.LogLevel;
     }
 
-    public void Log(string message) {
+    public void LogInfo(string message) {
+        if (_logLevel > LogLevel.Info) {
```

Comment:
In each message I added a comparison with the log level so that users can control which information they want to see.